app/models/ems_event.rb

```diff
@@ -146,6 +146,34 @@ def self.process_vm_in_event!(event, options = {})
         event[:vm_name] ||= vm.name
       end
     end
+
+    event_type = event[:event_type]
+    # both event types are Rhev specific
+    if (event_type == 'USER_ADD_VM' || event_type == 'USER_ADD_VM_FINISHED_SUCCESS') && event[:vm_or_template_id].nil?
```

Comment:
To me it looks like this code is totally misplaced here. I mean this is an Event being normalized and I would not expect it to create DB objects. 

What happens when a refresh is running, while this Event is processed? I totally agree with @gmcculloug that this will lead to trouble.

How is vmware doing a targeted refresh? Maybe copy this? Or wait for :skull: refresh :stuck_out_tongue_winking_eye: